libs/wingsdk/src/target-sim/api.inflight.ts

```diff
@@ -162,7 +162,7 @@ export class Api
   }
 
   private async saveState(state: StateFileContents): Promise<void> {
-    await fs.promises.writeFile(
+    fs.writeFileSync(
```

Comment:
I'm not against this change, but a more foolproof way to fix the issue would be to make the readFile more resilient instead right? Like if the file is corrupt/unparseable then it should remove it and assume no state is available.

Reply:
Sure -- yeah I was trying to fix the data getting corrupted, but we could also fix how we read the corrupted data. I can add that in